packages/build/tests/time/tests.js

```diff
@@ -2,10 +2,14 @@
 
 const { version } = require('process')
 
+const { intercept, cleanAll } = require('@netlify/nock-udp')
 const test = require('ava')
 
 const { runFixture } = require('../helpers/main')
-const { startUdpServer } = require('../helpers/udp_server')
+
+test.after(() => {
+  cleanAll()
+})
 
 test('Does not send plugin timings if no plugins', async (t) => {
```

Comment:
Other thing to note: we do not want to mock `Socket` for other tests which might rely on it, i.e. this approach requires putting tests that need mocking together in the same file.

We previously had tests split up in many small files. Refactoring it to a few big files sped up the tests by 2x because Ava only loads 2 test files at once in CI (even when the parallel options are enabled, this is hardcoded in Ava's codebase). So, we might want to be careful when mocking global variables as it might encourage smaller test files, which might increase the overall tests duration for that reason.

However, this is not a big problem and this works here :+1:

Reply:
> Other thing to note: we do not want to mock Socket for other tests which might rely on it

Right now our interception throws an error whenever a unmocked path is used - https://github.com/JGAntunes/nock-udp/blob/main/src/main.js#L36 - since we always validate that the current mock has been used and the `buffers` contain the data we need, we could make this behaviour optional, so that any tests running in parallel that rely on any unmocked path would use the original `Socket.send`

Reply:
Sound good :+1:

Reply:
Addressed via https://github.com/netlify/nock-udp/pull/2